ste/mgr-JobMgr.go

```diff
@@ -313,6 +430,30 @@ func (jm *jobMgr) AddJobPart(partNum PartNumber, planFile JobPartPlanFileName, s
 	jm.setFinalPartOrdered(partNum, jpm.planMMF.Plan().IsFinalPart)
 	jm.setDirection(jpm.Plan().FromTo)
 	jpm.exclusiveDestinationMap = jm.getExclusiveDestinationMap(partNum, jpm.Plan().FromTo)
+
+	jm.initMu.Lock()
+	defer jm.initMu.Unlock()
+
+	if jm.initState == nil {
+		jm.initState = &jobMgrInitState{}
+
+		// For UDAM to be initialized, there are some conditions:
+		// 1) The source must not be public (It's unsafe to send the OAuth token to a public source)
+		// 2) It should be a S2S transfer from blob (There's no need for a SAS on a download, and user delegation doesn't work on files)
+		// 3) The source should lack a SAS token (There's no need for SAS generation when we already have)
+		if !jpm.Plan().SourcePublic && jpm.Plan().FromTo.IsS2S() && jpm.Plan().FromTo.From() == common.ELocation.Blob() && len(jpm.sourceSAS) == 0 {
+			// Upgrade the default version for this transfer.
+			// Note that S2S transfers with a User delegation SAS will fail on a lower revision.
+			DefaultServiceApiVersion = "2018-11-09"
```

Comment:
>DefaultServiceApiVersion = "2018-11-09" [](start = 3, length = 39)

Oh, we need the more recent service version not to create the user delegation SAS, but to USE it?  Have I understood that correctly.  If so, maybe my earlier suggestion not to change the defualt globally was a bad idea.  I think I'd perfer a global change as you origienally did it to changing it here like this.